src/app/client/exercises/exercise-list/exercise-list.component.ts

```diff
@@ -17,28 +22,56 @@ export class ExerciseListComponent implements OnInit {
   exercisesFromClient: ExerciseModel[];
 
   constructor(private exerciseService: ExerciseService,
-              private youtubeService: YoutubeService) {
+              private youtubeService: YoutubeService,
+              private authService: AuthService,
+              private clientService: ClientService) {
   }
 
   ngOnInit() {
     // Instantiation of array.
     this.exercisesFromClient = [];
-    // Gets all exercises from the service
-    this.exerciseService.getExercises().subscribe(exercises => {
-      for (const exercise of exercises) {
-        // Gets the videoId on the exercise
-        const videoId = this.youtubeService.getIdFromURL(exercise.videoUrl);
-        this.youtubeService.getVideoInformation(videoId)
-          .subscribe(result => {
-            // Gets the imgUrl from the youtubeService.
-            const ytResponse = result as YoutubeResponse;
-            exercise.imgUrl = ytResponse.items[0].snippet.thumbnails.default.url;
-            this.exercisesFromClient.push(exercise);
-          });
-      }
+
+    this.getLoggedInUser();
+  }
+
+  /**
+      Gets the uid from the logged in user.
+   */
+  private getLoggedInUser() {
+    const userId = this.authService.getUserId();
+    this.clientService.getCurrentClientById(userId).subscribe(user => {
+      const loggedInUser = user as ClientModel;
+      const rehabilitationPlan = loggedInUser.rehabilitationPlan;
+      this.getExercisesFromRehabilitationPlan(rehabilitationPlan);
     });
   }
 
+  /**
+   * Get exercises from RehabilitationPlan.
+   */
+  private getExercisesFromRehabilitationPlan(rehabilitationPlan: RehabilitationPlan) {
+    if (rehabilitationPlan.exerciseIds) {
+      // Run only if there is an exercise.
+      if (rehabilitationPlan.exerciseIds.length > 0) {
+        // Adds the exercises one by one.
+        rehabilitationPlan.exerciseIds.forEach(exerciseId => {
```

Comment:
Since this also appears in another component (DRY) perhaps we should refactor?
If we have a ExerciseService it could be moved there?